sdk/signalr/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/core/rest/_rest_py3.py

```diff
@@ -0,0 +1,716 @@
+            curr_char = text[idx]
+            next_char = None if idx == len(text) - 1 else text[idx + 1]
+            if curr_char == "\n":
+                lines.append(text[: idx + 1])
+                text = text[idx + 1: ]
+                break
+            if curr_char == "\r" and next_char == "\n":
+                # if it ends with \r\n, we only do \n
+                lines.append(text[:idx] + "\n")
+                text = text[idx + 2:]
+                break
+            if curr_char == "\r" and next_char is not None:
+                # if it's \r then a normal character, we switch \r to \n
+                lines.append(text[:idx] + "\n")
+                text = text[idx + 1:]
+                break
+            if next_char is None:
+                text = ""
+                last_chunk_of_text += text
+                break
+    if last_chunk_of_text.endswith("\r"):
+        # if ends with \r, we switch \r to \n
+        lines.append(last_chunk_of_text[:-1] + "\n")
+    elif last_chunk_of_text:
+        lines.append(last_chunk_of_text)
+    return lines
+
+
+class _StreamContextManagerBase:
+    def __init__(
+        self,
+        client: Union[_PipelineClient, _AsyncPipelineClient],
+        request: "HttpRequest",
+        **kwargs
+    ):
+        """Used so we can treat stream requests and responses as a context manager.
+
+        In Autorest, we only return a `StreamContextManager` if users pass in `stream_response` True
+
+        Actually sends request when we enter the context manager, closes response when we exit.
+
+        Heavily inspired from httpx, we want the same behavior for it to feel consistent for users
+        """
+        self.client = client
+        self.request = request
+        self.kwargs = kwargs
+
+    @abstractmethod
+    def close(self):
+        ...
+
+class _StreamContextManager(_StreamContextManagerBase):
+    def __enter__(self) -> "HttpResponse":
+        """Actually make the call only when we enter. For sync stream_response calls"""
+        pipeline_transport_response = self.client._pipeline.run(
+            self.request._internal_request,
+            stream=True,
+            **self.kwargs
+        ).http_response
+        self.response = HttpResponse(
+            request=self.request,
+            _internal_response=pipeline_transport_response
+        )
+        return self.response
+
+    def __exit__(self, *args):
+        """Close our stream connection. For sync calls"""
+        self.response.__exit__(*args)
+
+    def close(self):
+        self.response.close()
+
+class _AsyncStreamContextManager(_StreamContextManagerBase):
+    async def __aenter__(self) -> "AsyncHttpResponse":
+        """Actually make the call only when we enter. For async stream_response calls."""
+        if not isinstance(self.client, _AsyncPipelineClient):
+            raise TypeError(
+                "Only sync calls should enter here. If you mean to do a sync call, "
```

Comment:
I think this should be "Only _async_ calls..."?

Reply:
Good point